### PR TITLE
Sumo Logic Cloud SIEM integration

### PR DESCRIPTION
**Pull request by matslofva**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Sumo Logic Cloud SIEM integration

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No(?)

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by marcellarichmond** [comment]
@mchasepan

File: Packs/SumoLogic_Cloud_SIEM/pack_metadata.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "Sumo Logic Cloud SIEM",
+    "description": "Cloud SIEM with SOC analytics and automation.",
```
Comment (sangsoojun88) [inline Packs/SumoLogic_Cloud_SIEM/pack_metadata.json:3]: -> " Sumo Logic Cloud SIEM provides threat detection and incident response for modern IT environments. This content pack will allow you to apply automation to perform actual SOC analyst workflows.  Using this content pack you will be able to fetch Incidents via Insights, update status of an Insight, add items to match list, add Threat Intel Indicators to Threat Intel Sources, and so on."

**Review by dantavori — CHANGES_REQUESTED** [review]
The script file looks very good, however we have some more stuff to fix - see my notes.
In addition, add a unittest for every command, including fetch incidents - see [Jask UT](https://github1s.com/demisto/content/blob/master/Packs/Jask/Integrations/Jask/Jask_test.py) for reference.
* after you fix the yml file, please regenerate the readme file of the integration (using demisto-sdk generate-docs).

Once we fix these issues we can proceed to a demo.

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.yml [hunk 2]
```diff
@@ -0,0 +1,253 @@
+commonfields:
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.yml:1]: Thanks, that should have been part of my last commit but wasn't for some reason. Updated.

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+    Convert last-seen argument to querystring
+    '''
+    if not argval or argval == 'All time':
+        return q
+    if argval == 'Last week':
+        return add_to_query(q) + '{}:NOW-7D..NOW'.format(timefield)
+    if argval == 'Last 48 hours':
+        return add_to_query(q) + '{}:NOW-48h..NOW'.format(timefield)
+    if argval == 'Last 24 hours':
+        return add_to_query(q) + '{}:NOW-24h..NOW'.format(timefield)
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py:81]: Ah predefined values for that arg should have been part of a previous commit to add `output` to the yml... Fixed.

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py [hunk 4]
```diff
@@ -0,0 +1,765 @@
...
+def insight_get_details(client: Client, args: Dict[str, Any]) -> CommandResults:
+    '''
+    Get insight details
+    '''
+    insight_id = args.get('insight-id')
+    if not insight_id:
+        raise ValueError('insight-id not specified')
+
+    resp_json = client.req('GET', 'sec/v1/insights/{}'.format(insight_id), {'exclude': 'signals.allRecords'})
+    insight = insight_signal_to_readable(resp_json['data'])
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py:196]: Correct - when the response status is ok (200) it is guaranteed to have `data` for most endpoints. A search that returns no results will also contain `data`:  `{'data': {'hasNextPage': False, 'objects': [], 'total': 0}, 'errors': []}`. I'll move `.get('data')` into the Client to be safe anyway.

File: Packs/SumoLogic_Cloud_SIEM/Integrations/ReleaseNotes/1_0_0.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#### Integrations
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/ReleaseNotes/1_0_0.md]: done

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py [hunk 6]
```diff
@@ -0,0 +1,765 @@
...
+        last_fetch = int(last_fetch)
+
+    # for type checking, making sure that latest_created_time is int
+    latest_created_time = cast(int, last_fetch)
+
+    # Initialize an empty list of incidents to return
+    # Each incident is a dict with a string as a key
+    incidents: List[Dict[str, Any]] = []
+
+    q = 'timestamp:>{}'.format(latest_created_time)
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py]: done

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py [hunk 7]
```diff
@@ -0,0 +1,765 @@
...
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client.req('GET', 'sec/v1/insights', {})
+        message = 'ok'
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py:201]: done

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py]: done

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py [hunk 8]
```diff
@@ -0,0 +1,765 @@
...
+        else:
+            raise e
+    return message
+
+
+def insight_get_details(client: Client, args: Dict[str, Any]) -> CommandResults:
+    '''
+    Get insight details
+    '''
+    insight_id = args.get('insight-id')
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/SumoLogic_Cloud_SIEM.py]: done

File: Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/README.md [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+There is no context output for this command.
+
+#### Command Example
+`!sumologic-sec-signal-search query="timestamp:NOW-7D..NOW name:contains(\"Internal\")"`
+
+#### Human Readable Output
+Signals:
+|Id|Name|Entity|RuleId|Description|Severity|Stage|Timestamp|ContentType|Tags|
+|---|---|---|---|---|---|---|---|---|---|
+| b50fd570-341b-576d-85b5-8b5cd17c0aee | IP Address Scan - Internal | 1.2.3.4 | LEGACY-S00050 | A scan of IP addresses | 3 | Discovery | 2021-04-22T04:08:13.514000 | RULE | _mitreAttackTactic:TA0007,<br>_mitreAttackTactic:TA0043,<br>_mitreAttackTechnique:T1046,<br>_mitreAttackTechnique:T1595 |
```
[the remark this answers is not in the document]
Comment (matslofva) [reply Packs/SumoLogic_Cloud_SIEM/Integrations/SumoLogic_Cloud_SIEM/README.md]: done

**Review by dantavori — APPROVED** [review]
great work!

**Pull request merged**